### ISSUE TITLE
Is the project dead?

### ISSUE THREAD

**Issue by nodiaque** [maurice-daly/DriverAutomationTool#515]
Hello,

I'm wondering if there's still development on that project? It seems to be halt, same with many scripts used by this solution.

**Comment by maurice-daly** [comment]
Hi,

The project isn't dead as such, it just has been a struggle to maintain of late due to lots of reasons. A new version of the tool and updated support in the driver/bios scripts to cater for Windows 11 23H2 will be out at the end of the week. 

Ideally I would like to completely re-code the solution, but work/life balance is difficult and I want to keep the solution free, so time is a bit limited at present. 

I hope you understand and please wait on the upcoming release.

**Comment by nodiaque** [comment]
Hello,

Thank you for the answer. I was hoping a reply of this type and it's all good. We all have life and this is all voluntary work in the end.

I was thinking on coding a new one myself since I wasn't seeing any update anywhere thus why I was wondering if I was taking the time or not.

Thanks for your hard work

**Comment by mgorski10** [comment]
Hi Maurice,

Along with the upcoming release, do you have plans to include an update to the Invoke-CMApplyDriverPackage script to take into account the deprecation of the AzureAD Powershell module?  I'm am working on a win10 to win11 migration and I'd like to precache the win11 drivers with the upgrade (over my CMG) but the depreciation has broken the process.  I am trying to see if I can use the Graph module in the script but I'm not sure where to begin and have not been successful.

Also, thanks for taking the time to work on this tool. It is extremely helpful.

**Comment by Latiron** [comment]
Maurice, can you also fix the Microsoft download link?  Looks like Microsoft has changed their download site info again and it has broken the ability to download Surface Driver packages.  Thank you.

**Comment by ajn142attamu** [comment]
> Hi,
> 
> The project isn't dead as such, it just has been a struggle to maintain of late due to lots of reasons. A new version of the tool and updated support in the driver/bios scripts to cater for Windows 11 23H2 will be out at the end of the week.
> 
> Ideally I would like to completely re-code the solution, but work/life balance is difficult and I want to keep the solution free, so time is a bit limited at present.
> 
> I hope you understand and please wait on the upcoming release.

If I can offer my unsolicited opinion (worth exactly what you paid for it), I think you're doing great work, but you're also doing too much. You've got a community of folks here happy to use your tool, and capable of working around things when they don't function as planned. I think it would be a good thing to leverage that by empowering the community to do some things like triage issues, consolidate them (how many are folks asking for the same thing at different times), and prioritize pull requests.

Eventually you build up a base of folks you trust to actually merge in changes to the tool itself to fix those issues, based on their understanding and the work they put in patching it for themselves. As part of that, I think you'd either need to document how you're packing the .ps1 source code down to an EXE, or alternatively, stop offering the .exe and just have folks work with the script directly. At that point, you're not having to spend the limited time you have for DAT on small bug fixes or simple feature requests, but rather on where you want it to go, or even on the rewrite.

**Comment by MikhailCompo** [comment]
Hi @maurice-daly Would you consider making the project open-source so it can be developed by others? Appreciate you haven't the time to work on it, perhaps agree to delegate some control?

**Comment by nodiaque** [comment]
I must say, having more person that could do the merge would be nice. In the meantime, people could do fork. Not the best thing but I think it's the way for now

**Comment by ajn142attamu** [comment]
> I must say, having more person that could do the merge would be nice. In the meantime, people could do fork. Not the best thing but I think it's the way for now

Agree completely RE forking, and that's what I've already done, but there's so many dang forks and it'd be nice if the effort each of us puts into solving problems solved them for more than just us, and helped the whole project move forward.

**Comment by nodiaque** [comment]
One thing is looking at the source, it seems to be a rebuild file from a project. I though it might be a Powershell Sapien but no. What tell me that is the region created for different file like startup.pss, global.ps1, mainform.psf and such. Would be great if the real source was dumped in github and not the compile ps1 file.

**Comment by ajn142attamu** [comment]
@nodiaque doesn't look like that's happening 😆

**Comment by nodiaque** [comment]
Yeah, I also understood why the source code doesn't recompile the project in sapiens, the recovery code isn't included like it's by default. Would be great to properly put all source in an opensource project...

**Comment by maurice-daly** [comment]
Quick update. I hear a lot of people asking for this to be open source, here is my take.

I have spent a lot of time creating the solution, and I get that my availability to maintain the solution has for some caused concern, especially for those who rely on it. Remember this is a free solution, that I have put a lot of time in, but I do hear you.

At the present time, in its current state, it will not be open source. I have had huge issues with security vulnerability scanners of late, and even distributions of the tool with key gens, that are obviously malware. Due to this I do need some "control".

With this said, I feel I need to rebuild from the group up. There are drivers (no pun intended) that are pushing me to drop support for some products (MDT) and add support for others that we thought would be handled natively (Intune).

What I can say is that I appreciate the offer of assistance, so once a V8 appears (late April timeframe hopefully), I will be opening up support from the community. I'll provide more information at that point, but the key thing is, I appreciate you and many hands make light work.

Thanks for the patience and for the support from everyone.
